replica/database.hh

```diff
@@ -842,7 +842,7 @@ public:
     const locator::effective_replication_map_ptr& get_effective_replication_map() const { return _erm; }
     future<> update_effective_replication_map(locator::effective_replication_map_ptr);
     [[gnu::always_inline]] bool uses_tablets() const;
-    future<> cleanup_tablet(database&, db::system_keyspace&, locator::tablet_id);
+    future<> cleanup_tablet(database&, db::system_keyspace&, locator::tablet_id, bool deallocate_sg = true);
```

Comment:
Deallocating should be a separate function. Something as critical as deallocating the storage group shouldn't be hidden behind a default parameter.